nautobot/extras/filters/__init__.py

```diff
@@ -448,10 +452,44 @@ class NautobotFilterSet(
 
 
 class ContactFilterSet(NameSearchFilterSet, NautobotFilterSet):
+    similar_contacts = NaturalKeyOrPKMultipleChoiceFilter(
```

Comment:
Let's be more explicit and make this something like `similar_to_location_data` or the like?